browser/brave_content_browser_client.cc

```diff
@@ -561,13 +567,19 @@ void BraveContentBrowserClient::
         content::RenderFrameHost& render_frame_host,                // NOLINT
         blink::AssociatedInterfaceRegistry& associated_registry) {  // NOLINT
 #if BUILDFLAG(ENABLE_WIDEVINE)
+// We only need the DRM Tab Helper when Widevine is available through component
+// updates. This is the case on x64 Linux. On other architectures such as Arm64,
+// ENABLE_WIDEVINE is still set to let users supply Widevine themselves.
+// Only use the DRM Tab Helper when it is actually needed:
+#if !BUILDFLAG(IS_LINUX) || defined(ARCH_CPU_X86_64)
   associated_registry.AddInterface<
       brave_drm::mojom::BraveDRM>(base::BindRepeating(
       [](content::RenderFrameHost* render_frame_host,
          mojo::PendingAssociatedReceiver<brave_drm::mojom::BraveDRM> receiver) {
         BraveDrmTabHelper::BindBraveDRM(std::move(receiver), render_frame_host);
       },
       &render_frame_host));
+#endif
```

Comment:
ditto - how about using something like `bool NeedBraveDrmTabHelper()` util method?
Then, we could remove most of ifdefs from source and gn files.